src/python/zquantum/core/circuits/_generators.py

```diff
@@ -17,28 +17,53 @@ def create_layer_of_gates(
 
     Args:
         number_of_qubits: number of qubits in the circuit
-        gate_name: the single qubit gate to be applied to each qubit
+        gate_factory: the single qubit gate to be applied to each qubit
         parameters: parameters of the single-qubit gates
 
     Returns:
         circuit: Created circuit.
     """
     circuit = Circuit()
 
+    return apply_gate_to_qubits(
+        circuit, range(number_of_qubits), gate_factory, parameters
+    )
+
+
+def apply_gate_to_qubits(
+    circuit: Circuit,
+    qubit_indices: Iterable[int],
+    gate_factory: Union[GatePrototype, Gate],
+    parameters: Optional[np.ndarray] = None,
+) -> Circuit:
+    """Apply the passed gate to the specified qubits
+
+    Args:
+        circuit: circuit to add the gate to
+        qubit_indices: list of qubits that to apply the gate on,
+            duplicates will be ignored
```

Comment:
Ignoring duplicates seem to be reasonable, however, users might still accidentally pass duplicates.

Hence, I would suggest raising an exception, or at the very least a warning if duplicates are present.

Reply:
Done! :)